### DISCUSSION TITLE
Make the current page pathname available in middleware on NextRequest

### DISCUSSION THREAD

**Discussion by KurtRogers** [vercel/next.js#63777] [Ideas]
### Goals

1. Enable developers to get the current page pathname in the middleware

### Background

Somehow managed to implement dynamic redirects by making a strict regex in the matcher and also disabling the next link processing in the `missing` object as recommended by the docs here:
https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy

It would've been much easier if I had the current page pathname available to me in Next JS (or some guidance on how to make this available in my project.)

Currently the matcher is doing a good job of ensuring that only the current url is the only item returned. But it isn't as reassuring as I'd like it to be.

```js
  const pathname = request.nextUrl;

  const { href, pathname: _path } = pathname;

  // check that the page exists and 200's
  const { status: pageStatusResult } = await fetch(href);

  // if page exists, stop everything here
  if (pageStatusResult === 200) {
    return NextResponse.next();
  }

  // run the redirection functionality passing `_path`
```

### Proposal

Provide the current page path in the request object for middleware. 

I'd be more than happy to help where I can on this.

**Comment by Netail** [comment]
You can simplify the check if your page exist to:
```ts
const res = NextResponse.next();

if (res.ok) {
    return res;
} else {
    return NextResponse.redirect('/redirect-path');
}
```

**Reply by KurtRogers** [reply]
Thanks Netail, I'll implement that in my work. 

I'm still stuck on being able to accurately grab the next route pathname; I'm doing a check against my redirect API:
```js
    const redirects = await fetch(
      getFrontendUrl(`/api/redirects?filters[from]=${_route pathname var_}`)
    );
```

This is only called IF the page doesn't exist, it returns the redirect for the specific path. But something I didn't realise was that middleware is ran for every prefetch meaning that the next/link has become a bit of a poison chalice; great for user experience but in reality potentially terrible for performance and makes it almost impossible to implement something perfect for middleware - like **dynamic redirects**

**Reply by Netail** [reply]
I believe prefetches do have a special header, which you could filter out
